eZ/Publish/API/Repository/Tests/SearchEngineIndexingTest.php

```diff
@@ -292,16 +292,84 @@ public function testDeleteLocation()
+
+        $content02 = $this->createContentWithName('content02', [2]);
+        $location02 = $locationService->loadLocation($content02->contentInfo->mainLocationId);
+
+        $locationService->swapLocation($location01, $location02);
+        $this->refreshSearch($repository);
+
+        // content02 should be at location01
+        $criterion = new Criterion\LocationId($location01->id);
+        $query = new Query(['filter' => $criterion]);
+        $results = $searchService->findContent($query);
+        $this->assertEquals(1, $results->totalCount);
+        $this->assertEquals($content02->id, $results->searchHits[0]->valueObject->id);
+
+        // content01 should be at location02
+        $criterion = new Criterion\LocationId($location02->id);
+        $query = new Query(['filter' => $criterion]);
+        $results = $searchService->findContent($query);
+        $this->assertEquals(1, $results->totalCount);
+        $this->assertEquals($content01->id, $results->searchHits[0]->valueObject->id);
+    }
+
+    /**
+     * Test that updating Content metadata affects properly Search Engine Index.
+     */
+    public function testUpdateContentMetadata()
+    {
+        $repository = $this->getRepository();
+        $contentService = $repository->getContentService();
+        $locationService = $repository->getLocationService();
+        $searchService = $repository->getSearchService();
+
+        $publishedContent = $this->createContentWithName('updateMetadataTest', [2]);
+        $newLocationCreateStruct = $locationService->newLocationCreateStruct(60);
+        $newLocation = $locationService->createLocation($publishedContent->contentInfo, $newLocationCreateStruct);
+
+        $newContentMetadataUpdateStruct = $contentService->newContentMetadataUpdateStruct();
+        $newContentMetadataUpdateStruct->remoteId = md5('Test');
+        $newContentMetadataUpdateStruct->publishedDate = new \DateTime('+10m');
+        $newContentMetadataUpdateStruct->mainLocationId = $newLocation->id;
+
+        $contentService->updateContentMetadata($publishedContent->contentInfo, $newContentMetadataUpdateStruct);
+        $this->refreshSearch($repository);
+
+        $criterion = new Criterion\ContentId($publishedContent->id);
+        $query = new Query(['filter' => $criterion]);
+        $results = $searchService->findContent($query);
+        $this->assertEquals(1, $results->totalCount);
+        $foundContent = $results->searchHits[0]->valueObject;
+        /** @var \eZ\Publish\Core\Repository\Values\Content\Content $foundContent */
+        $this->assertEquals($publishedContent->id, $foundContent->id);
+        $this->assertEquals($newContentMetadataUpdateStruct->publishedDate, $foundContent->contentInfo->publishedDate);
+        $this->assertEquals($newLocation->id, $foundContent->contentInfo->mainLocationId);
+        $this->assertEquals($newContentMetadataUpdateStruct->remoteId, $foundContent->contentInfo->remoteId);
```

Comment:
These asserts should fail, because UpdateContentMetadata Search Slot does not exist, however [it passes](https://travis-ci.org/ezsystems/ezpublish-kernel/jobs/160149888#L832). Tried to debug this but can't find a point when it is indexed with updated values.
ping @pspanja